bin/weewx/drivers/ws1.py

```diff
@@ -51,26 +55,45 @@ def logerr(msg):
 
 class WS1Driver(weewx.drivers.AbstractDevice):
     """weewx driver that communicates with an ADS-WS1 station
-    
-    port - serial port
-    [Required. Default is /dev/ttyS0]
+
+    mode - Communication mode - TCP, UDP, or Serial
+    [Required. Default is serial]
+
+    port - serial port or TCP address
+    [Required. Default is /dev/ttyS0 for serial, and 192.168.36.25:3000 for TCP]
 
     max_tries - how often to retry serial communication before giving up
     [Optional. Default is 5]
 
     retry_wait - how long to wait, in seconds, before retrying after a failure
     [Optional. Default is 10]
+
+    timeout - The amount of time, in seconds, before the connection fails if
+    there is no response
+    [Optional. Default is 3]
     """
     def __init__(self, **stn_dict):
-        self.port = stn_dict.get('port', DEFAULT_PORT)
+        con_mode = stn_dict.get('mode', 'serial').lower()
+        if con_mode == 'serial':
+            self.port = stn_dict.get('port', DEFAULT_SER_PORT)
+        elif con_mode == 'tcp' or con_mode == 'udp':
+            self.port = stn_dict.get(
+                'port', '%s:%d' % (DEFAULT_TCP_ADDR, DEFAULT_TCP_PORT))
+        else:
+            # exit(3)
+            pass
```

Comment:
consider this instead:

if con_mode == 'tcp' or con_mode == 'udp':
    self.port = ...
else:
    self.port = stn_dict.get('port', DEFAULT_SER_PORT)

if you do not like that kind of quiet backward compatibility, then fail hard with an exception indicating that 'mode' is a required parameter.